File: backend/app/models/customer.py
from typing import TYPE_CHECKING

from sqlalchemy import String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base

if TYPE_CHECKING:
    from app.models.reservation import Reservation


class Customer(Base):
    __tablename__ = "customers"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    full_name: Mapped[str] = mapped_column(String(100), nullable=False)
    phone_number: Mapped[str | None] = mapped_column(String(20), nullable=True)
    preferred_contact_method: Mapped[str] = mapped_column(String(50), nullable=False)
    messenger_user_name: Mapped[str | None] = mapped_column(String(100), nullable=True)
    notes: Mapped[str | None] = mapped_column(Text, nullable=True)


reservations: Mapped[list["Reservation"]] = relationship(
    back_populates="customer",
)

File: frontend/src/types/customer.ts
export interface Customer {
  id: number;
  full_name: string;
  phone_number: string | null;
  preferred_contact_method: string;
  messenger_user_name: string | null;
  notes: string | null;
}

export interface CustomerCreate {
  full_name: string;
  phone_number: string | null;
  preferred_contact_method: string;
  messenger_user_name: string | null;
  notes: string | null;
}